ff/src/fields/models/fp.rs

```diff
@@ -0,0 +1,967 @@
+use ark_std::{
+    cmp::{Ord, Ordering, PartialOrd},
+    fmt::{Display, Formatter, Result as FmtResult},
+    io::{Read, Result as IoResult, Write},
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    str::FromStr,
+};
+use num_traits::{One, Zero};
+
+use crate::{
+    biginteger::{arithmetic as fa, BigInt, BigInteger as _BigInteger},
+    bytes::{FromBytes, ToBytes},
+    fields::{FftField, Field, FpParameters, LegendreSymbol, PrimeField, SquareRootField},
+};
+use ark_serialize::*;
+
+pub trait FpParams<const N: usize>: FpParameters<BigInt = BigInt<N>> {}
```

Comment:
I think we can get rid of this trait now, no?

Reply:
Well, for now, I use it to instantiate NO_CARRY as a const param. Since otherwise, it was troublesome to ensure it would evaluate at compile time.